lib/active_model/serializer/builder.rb

```diff
@@ -0,0 +1,71 @@
+require "set"
+module ActiveModel
+  class Serializer
+    class Builder
```

Comment:
Do you have a suggestion for a better name?  I believe AMS used to have a build_as_json method, or something like that.  I actually don't like builder all that much either, but honestly, I find the adapter being what we call serializable_hash on, rather than the serializer, more confusing, which is part of why I'd like to encapsulate their combined behavior.

(I'm actually thinking of collecting common class names from Ruby projects and seeing what works and what doesn't. This is of interest to me as well).